common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/ExternalShuffleBlockResolver.java

```diff
@@ -172,7 +172,7 @@ public ManagedBuffer getBlockData(
       String appId,
       String execId,
       int shuffleId,
-      int mapId,
+      long mapId,
```

Comment:
@xuanyuanking why change this from int to long? Is it possible that a mapId can be greater than 2^31?

Reply:
previous the map id is the index of the mapper, and can get conflicts when we re-run the task. Now the map id is the task id, which is unique. task id needs to be long.

Reply:
Yes, after this patch, we set mapId by using the `taskAttemptId` of map task, which is a unique Id within the same SparkContext. You can see the comment https://github.com/apache/spark/pull/25620#discussion_r319396089